test/integration/raw-queries.test.js

```diff
@@ -0,0 +1,266 @@
+        let query = 'INSERT INTO dummies (id, name) VALUES($id, \'abc$$\');';
+
+        if (dialect === 'mssql') {
+          query = `SET IDENTITY_INSERT dummies ON; ${query}; SET IDENTITY_INSERT dummies OFF;`;
+        }
+
+        return sequelize.query(query, {
+          type: sequelize.QueryTypes.INSERT,
+          bind: {
+            id: 20
+          }
+        });
+      }).then(() => {
+        return sequelize.query('SELECT * FROM dummies WHERE id = $id AND name = \'abc$$\';', {
+          model: Model,
+          type: sequelize.QueryTypes.SELECT,
+          bind: {
+            id: 20
+          }
+        });
+      }).then(instances => {
+        expect(instances.length).to.equal(1);
+        expect(instances[0].get({ plain: true })).to.deep.equal({
+          id: 20,
+          name: 'abc$'
+        });
+      });
+    });
+
+  });
+
+
+  describe('Compositions', () => {
+
+    it('can use compositions', () => {
+      const Model = createModel();
+
+      return createTable().then(() => {
+        // Nested composition just for testing, otherwise it is unneeded
+        let query = sequelize.composition('INSERT INTO dummies (id, name) ', sequelize.composition('VALUES(', sequelize.slot(20)), ', ', sequelize.slot('abc\'"\n\r\b\t\\\x1a'), ');');
+
+        if (dialect === 'mssql') {
+          query = sequelize.composition('SET IDENTITY_INSERT dummies ON; ', query, '; SET IDENTITY_INSERT dummies OFF;');
+        }
+
+        return sequelize.query(query, {
+          type: sequelize.QueryTypes.INSERT
+        });
+      }).then(() => {
+        return sequelize.query(sequelize.composition('SELECT * FROM dummies WHERE name = ', sequelize.slot('abc\'"\n\r\b\t\\\x1a'), ';'), {
+          model: Model,
+          type: sequelize.QueryTypes.SELECT
+        });
+      }).then(instances => {
+        expect(instances.length).to.equal(1);
+        expect(instances[0].get({ plain: true })).to.deep.equal({
+          id: 20,
+          name: 'abc\'"\n\r\b\t\\\x1a'
+        });
+      });
+    });
+
+    it('do not need to escape $', () => {
+      const Model = createModel();
+
+      return createTable().then(() => {
+        // Nested composition just for testing, otherwise it is unneeded
+        let query = sequelize.composition('INSERT INTO dummies (id, name) ', sequelize.composition('VALUES(', sequelize.slot(20)), ', \'abc$\');');
+
+        if (dialect === 'mssql') {
+          query = sequelize.composition('SET IDENTITY_INSERT dummies ON; ', query, '; SET IDENTITY_INSERT dummies OFF;');
+        }
+        
+        return sequelize.query(query, {
+          type: sequelize.QueryTypes.INSERT
+        });
+      }).then(() => {
+        return sequelize.query(sequelize.composition('SELECT * FROM dummies WHERE id = ', sequelize.slot(20), ' AND name = \'abc$\';'), {
+          model: Model,
+          type: sequelize.QueryTypes.SELECT
+        });
+      }).then(instances => {
+        expect(instances.length).to.equal(1);
```

Comment:
Please use `expect(...).to.have.lengthOf(...)`, this show show a nicer diff on failure :)

Reply:
Yes, totally missed that. Now it's solved.